pcsx2/gui/MainMenuClicks.cpp

```diff
@@ -412,6 +412,26 @@ void MainEmuFrame::Menu_IsoBrowse_Click( wxCommandEvent &event )
 	AppSaveSettings();		// save the new iso selection; update menus!
 }
 
+void MainEmuFrame::Menu_IsoClear_Click(wxCommandEvent &event)
+{
+	wxDialogWithHelpers dialog(this, _("Confirm clearing ISO list"));
+	dialog += dialog.Heading(_("This will clear your entire ISO list. Continue?"));
+
+	if (SysHasValidState())
+		dialog += dialog.Heading(_("(Your running ISO will remain in the list)"));
+
+	bool confirmed = (pxIssueConfirmation(dialog, MsgButtons().Yes().Cancel()) != wxID_CANCEL);
+
+	if (confirmed)
+	{
+		if (!SysHasValidState())
```

Comment:
If you boot with the `Plugin` or `No disc` option selected then there'll still be one entry left in the recent ISO list even though an ISO isn't running.